### PR TITLE
Added initial implementation of pysimplegui

### PR DESCRIPTION
**Pull request by polo123qwe**
As for the issue #6, I have created a simple interface with the pysimplegui library.

I've added the option of still using the cli version if you run the program with the `-cli` argument.

I could not fully test it since I do not have a working account for the site, @OtchereDev could you give it complete test to see if it works as expected?

Edit: I also did not manage to find a way to get a download %, how should this be done?

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,93 +1,40 @@
-from module.create import create_folder, create_logger
```
Comment (OtchereDev) [inline main.py:1]: I like the idea of allowing the user to run either cli or the gui....Great work 👍

File: module/gui/render_gui.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import PySimpleGUI as sg
```
Comment (OtchereDev) [inline module/gui/render_gui.py]: The main idea of the GUI is to move completely from the cli (to make it a stand alone). But in this situation it still rely on the cli to make the download....Anyways i like the simplicity of the dashboard 👍

Comment (OtchereDev) [reply module/gui/render_gui.py]: It is suppose to include the download mode which is not taken at the initial start of the app

**Comment by OtchereDev** [comment]
if you want to test out your work use the following details:
Email: pass
Password: pass
Course_link : https://coursehunter.net/course/python-django-dev-to-deployment

**Comment by OtchereDev** [comment]
**Edit: I also did not manage to find a way to get a download %, how should this be done?**

with the question above, i was think of destroying the window it has login, then show a list of all the videos in the course and allow the user to select the ones he/she want then using pysimplegui popup to show the downloading.

**Comment by OtchereDev** [comment]
check this for some ideas and inspiration : https://github.com/PySimpleGUI/PyIDM

thanks for helping 👍

**Comment by polo123qwe** [comment]
@OtchereDev I believe something like this is what you had in mind?

File: main.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 from module.process import download
```
Comment (OtchereDev) [inline main.py:2]: Excellent work 🥇.... But there are a few problems:
1. The progress bar work but it does not write the download to the file. As in after the video is indicated by the progress bar as done, the total size of the video file is just 1KB which is not suppose to be so same for the course materials.
2. When a download is ongoing and you click on the download btn again, then some fuzzy clitches happens on progress bar
3. Try and beautify the UI a bit

Overall that is a great work you have done 👍

Comment (polo123qwe) [reply main.py:2]: About 1. I believe when I tested it, the file was downloaded properly, I'll recheck that.
The 2nd one the download button is supposed to be disabled when a download is ongoing, if its not then I have to fix that
3. I will try to give it some styles, but I'm not very good at the frontend.

**Comment by OtchereDev** [comment]
send
this is what happened when i tried a download...check the file size

File: module/extractor/download_iterator.py [hunk 4]
```diff
@@ -11,8 +11,7 @@ def download_and_save_iterator(r, chunk_size, total_size, title, isGUI=False, wi
                 download_percentage = ((chunk_size * (curr_chunk + 1)) / total_size) * 100
```
Comment (OtchereDev) [inline module/extractor/download_iterator.py:11]: Exxxxcccceeellllleeeennntttt work 👍 💯, I have not tested it absolutely but the download for the course materials works perfectly, I will do the final testing at later time. But in the mean time, I want you to make some few changes:
1. Inside the extract_video_url.py, on line 14 add ```parser="lxml" ``` inside the Robobrowser function to make it ``` br = Robobrowser(parser="lxml") ```
2. In the same file on line 54 change the word ```vast``` to ```poster```

File: module/extractor/download_iterator.py [hunk 5]
```diff
@@ -11,8 +11,7 @@ def download_and_save_iterator(r, chunk_size, total_size, title, isGUI=False, wi
...
                 window[element['progress_bar_key']].update(download_percentage)
                 window[element['percent_key']].update(download_percentage)
-
-            f.write(data)
+                f.write(data)
```
Comment (OtchereDev) [inline module/extractor/download_iterator.py:14]: Secondly I have created a branch for the GUI, so please make a new pr to that branch for me when you do with change please.
Lastly, do you know about the module py2exe or pyinstaller, generate single .exe file

I need your email or whatsapp for future collaboration 🥂

Thank you

Comment (polo123qwe) [reply module/extractor/download_iterator.py:14]: I have never worked with either of those modules before. My email is: bernausergi@gmail.com

**Comment by polo123qwe** [comment]
I've updated the base branch so the PR points to the GUI one

**Comment by OtchereDev** [comment]
@polo123qwe if incase you want to download a course  from coursehunter which is premium let me know, i will shoot you my login details through your email.....

**Comment by polo123qwe** [comment]
Thanks! Also could you add the label `hacktoberfest-accepted` to the PR, it would really help me.

**Pull request merged**